cmd/zt_credentialUtil_test.go

```diff
@@ -34,46 +34,50 @@ var _ = chk.Suite(&credentialUtilSuite{})
-		{common.ECredentialType.OAuthToken(), "http://myaccount.blob.core.core.usgovcloudapi.net", "", true},
-		{common.ECredentialType.SharedKey(), "http://myaccount.dfs.core.windows.net", "", true},
-		{common.ECredentialType.S3AccessKey(), "http://something.s3.eu-central-1.amazonaws.com", "", true},
-		{common.ECredentialType.S3AccessKey(), "http://something.s3.cn-north-1.amazonaws.com.cn", "", true},
-		{common.ECredentialType.S3AccessKey(), "http://s3.eu-central-1.amazonaws.com", "", true},
-		{common.ECredentialType.S3AccessKey(), "http://s3.cn-north-1.amazonaws.com.cn", "", true},
-		{common.ECredentialType.S3AccessKey(), "http://s3.amazonaws.com", "", true},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://myaccount.blob.core.windows.net", "", true},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://myaccount.blob.core.chinacloudapi.cn", "", true},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://myaccount.blob.core.cloudapi.de", "", true},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://myaccount.blob.core.core.usgovcloudapi.net", "", true},
+		{common.ECredentialType.SharedKey(), common.ELocation.BlobFS(), "http://myaccount.dfs.core.windows.net", "", true},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://something.s3.eu-central-1.amazonaws.com", "", true},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://something.s3.cn-north-1.amazonaws.com.cn", "", true},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://s3.eu-central-1.amazonaws.com", "", true},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://s3.cn-north-1.amazonaws.com.cn", "", true},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://s3.amazonaws.com", "", true},
 
 		// These should fail (they are not storage)
-		{common.ECredentialType.OAuthToken(), "http://somethingelseinazure.windows.net", "", false},
-		{common.ECredentialType.S3AccessKey(), "http://somethingelseinaws.amazonaws.com", "", false},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://somethingelseinazure.windows.net", "", false},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://somethingelseinaws.amazonaws.com", "", false},
 
 		// As should these (they are nothing to do with the expected URLs)
-		{common.ECredentialType.OAuthToken(), "http://example.com", "", false},
-		{common.ECredentialType.S3AccessKey(), "http://example.com", "", false},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://example.com", "", false},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://example.com", "", false},
+		// Test that we don't want to send an S3 access key to a blob resource type.
+		{common.ECredentialType.S3AccessKey(), common.ELocation.Blob(), "http://example.com", "", false},
```

Comment:
Should this url be a blob URL, if its trying to test (only) what the comment above it says?